Performance/QPSBenchmark/Sources/QPSBenchmark/Runtime/Async/AsyncQPSServerImpl.swift

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import GRPC
+import Logging
+import NIOCore
+import NIOPosix
+
+/// Server setup for asynchronous requests.
+final class AsyncQPSServerImpl: AsyncQPSServer {
+  private let eventLoopGroup: MultiThreadedEventLoopGroup
+  private let server: Server
+  private let threadCount: Int
+
+  private var statsPeriodStart: DispatchTime
+  private var cpuStatsPeriodStart: CPUTime
+
+  private let logger = Logger(label: "AsyncQPSServerImpl")
+
+  /// Initialisation.
+  /// - parameters:
+  ///     - config: Description of the type of server required.
+  init(config: Grpc_Testing_ServerConfig) async throws {
+    // Setup threads as requested.
+    let threadCount = config.asyncServerThreads > 0
+      ? Int(config.asyncServerThreads)
+      : System.coreCount
+    self.threadCount = threadCount
+    self.logger.info("Sizing AsyncQPSServerImpl", metadata: ["threads": "\(threadCount)"])
+    self.eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: threadCount)
+
+    // Start stats gathering.
+    self.statsPeriodStart = grpcTimeNow()
+    self.cpuStatsPeriodStart = getResourceUsage()
+
+    let workerService = AsyncBenchmarkServiceImpl()
+
+    // Start the server
+    self.server = try await Server.insecure(group: self.eventLoopGroup)
+      .withServiceProviders([workerService])
+      .withLogger(self.logger)
+      .bind(host: "localhost", port: Int(config.port))
+      .get()
+  }
+  
+  func getServerInfo() -> ServerInfo {
+    let port = self.server.channel.localAddress?.port ?? 0
+    return ServerInfo(threadCount: self.threadCount, port: port)
+  }
```

Comment:
This can just be a computed property ("getter" functions aren't very Swifty):

```suggestion
  var serverInfo: ServerInfo {
    let port = self.server.channel.localAddress?.port ?? 0
    return ServerInfo(threadCount: self.threadCount, port: port)
  }
```